fsc/hdf5_io/_save_load.py

```diff
@@ -12,6 +12,45 @@
 __all__ = ['save', 'load']
 
 
+@lru_cache(maxsize=None)
+def _get_entrypoint_mapping(group):
+    """
+    Helper function to get the entry point mapping corresponding to a
+    given group name
+    """
+    import pkg_resources
+    return {ep.name: ep for ep in pkg_resources.iter_entry_points(group=group)}
+
+
+def _try_loading_parts(*, identifier, entry_point_mapping):
+    """
+    Helper function to load an entrypoint corresponding to the given
+    identifier. If an exact match for the identifier is not found, the
+    colon-separated "parent identifiers" will be checked.
+
+    For example, for an entry point named 'nobody.expects.the',
+    it will check 'nobody.expects.the', 'nobody.expects', 'nobody', in
+    that order.
+
+    :param identifier: The identifier for which an entrypoint is to be loaded.
+    :type identifier: str
+
+    :param entry_point_mapping: A mapping between entrypoint names and the entrypoints themselves.
+    :type entry_point_mapping: dict
+
+    :returns: The name of the entry point that was loaded (if any), or False.
+    """
+    split_identifier = identifier.split('.')
```

Comment:
I would favor to kick out parts with leading _ (i.e. not mydict._whereever_it_resides.MyContainer, but mydict.MyContainer)
We want to track the class by "import name", not actual file name. If we track it exactly, refactoring will produce a cost (fixing entry points...) and therefore not be done.

Reply:
but maybe it's not pythonic... then leave it as is, if there is an implementation detail (in what exact submodule it is) in the entry points, its not too bad... And people will be remindend (latest on loading the entry point) to change it since it breaks the entry point if the class moves to another file

Reply:
In general I agree, it's better to use the "public" import name. But, if you want to use this to load an internal module only on demand, I think it should still be permissible to use an import with leading underscore.

We're also using this internally, because the `_sympy_load` and `_sympy_save` module should not be public.